src/Jackett.Common/Indexers/CardigannIndexer.cs

```diff
@@ -1463,39 +1472,65 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
+                                if (!LoginResult)
+                                    throw new Exception(string.Format("Relogin failed"));
+                                await TestLogin();
+                                response = await RequestWithCookiesAsync(searchUrl, method: method, data: queryCollection);
+                                if (response.IsRedirect && SearchPath.Followredirect)
+                                    await FollowIfRedirect(response);
+
+                                results = response.ContentString;
+                                SearchResultDocument = SearchResultParser.ParseDocument(results);
+                            }
                         }
 
                         checkForError(response, Definition.Search.Error);
 
-                        if (Search.Preprocessingfilters != null)
+                        IHtmlCollection<IElement> rowsDom;
+
+                        if (SearchPath.Response != null && SearchPath.Response.Type.Equals("xml"))
+                        {
+                            var SearchResultParser = new XmlParser();
+                            var SearchResultDocument = SearchResultParser.ParseDocument(results);
+
+                            if (search.Preprocessingfilters != null)
+                            {
+                                results = applyFilters(results, search.Preprocessingfilters, variables);
+                                SearchResultDocument = SearchResultParser.ParseDocument(results);
+                                logger.Trace(string.Format("CardigannIndexer ({0}): result after preprocessingfilters: {1}", Definition.Id, results));
+                            }
+
+                            var rowsSelector = applyGoTemplateText(search.Rows.Selector, variables);
+                            rowsDom = SearchResultDocument.QuerySelectorAll(rowsSelector);
+                        }
+                        else
                         {
-                            results = applyFilters(results, Search.Preprocessingfilters, variables);
-                            SearchResultDocument = SearchResultParser.ParseDocument(results);
-                            logger.Debug(string.Format("CardigannIndexer ({0}): result after preprocessingfilters: {1}", Id, results));
+                            var SearchResultParser = new HtmlParser();
+                            var SearchResultDocument = SearchResultParser.ParseDocument(results);
+
+                            if (search.Preprocessingfilters != null)
+                            {
+                                results = applyFilters(results, search.Preprocessingfilters, variables);
+                                SearchResultDocument = SearchResultParser.ParseDocument(results);
+                                logger.Trace(string.Format("CardigannIndexer ({0}): result after preprocessingfilters: {1}", Definition.Id, results));
```

Comment:
```suggestion
                                logger.Debug(string.Format("CardigannIndexer ({0}): result after preprocessingfilters: {1}", Definition.Id, results));
```
we use a different logging level than prowlerr